File: common/protcol/c#/caller/client_call_gatecaller.cs
/*this caller file is codegen by juggle for c#*/
using System;
using System.Collections;
using System.IO;

namespace caller
{
    public class client_call_gate : juggle.Icaller 
    {
        public client_call_gate(juggle.Ichannel _ch) : base(_ch)
        {
            module_name = "client_call_gate";
        }

        public void connect_server(String argv0,Int64 argv1)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            call_module_method("connect_server", _argv);
        }

        public void cancle_server()
        {
            ArrayList _argv = new ArrayList();
            call_module_method("cancle_server", _argv);
        }

        public void connect_hub(String argv0,String argv1)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            call_module_method("connect_hub", _argv);
        }

        public void disconnect_hub(String argv0,String argv1)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            call_module_method("disconnect_hub", _argv);
        }

        public void forward_client_call_hub(String argv0,String argv1,String argv2,ArrayList argv3)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            call_module_method("forward_client_call_hub", _argv);
        }

        public void heartbeats(Int64 argv0)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            call_module_method("heartbeats", _argv);
        }

    }
}

File: common/protcol/js/caller/client_call_gatecaller.js
/*this caller file is codegen by juggle for js*/
function client_call_gate_caller(ch){
    Icaller.call(this, "client_call_gate", ch);

    this.connect_server = function( argv0, argv1){
        var _argv = [argv0,argv1];
        this.call_module_method.call(this, "connect_server", _argv);
    }

    this.cancle_server = function(){
        var _argv = [];
        this.call_module_method.call(this, "cancle_server", _argv);
    }

    this.connect_hub = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "connect_hub", _argv);
    }

    this.enable_heartbeats = function(){
        var _argv = [];
        this.call_module_method.call(this, "enable_heartbeats", _argv);
    }

    this.disable_heartbeats = function(){
        var _argv = [];
        this.call_module_method.call(this, "disable_heartbeats", _argv);
    }

    this.forward_client_call_hub = function( argv0, argv1, argv2, argv3){
        var _argv = [argv0,argv1,argv2,argv3];
        this.call_module_method.call(this, "forward_client_call_hub", _argv);
    }

    this.heartbeats = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "heartbeats", _argv);
    }

}
(function(){
    var Super = function(){};
    Super.prototype = Icaller.prototype;
    client_call_gate_caller.prototype = new Super();
})();
client_call_gate_caller.prototype.constructor = client_call_gate_caller;

